### DISCUSSION TITLE
Find markers information such as Intro, verse, chorus etc from a song

### DISCUSSION THREAD

**Discussion by logesh2496** [MTG/essentia#1371] [Q&A]
We are looking to find the markers information from a music file using essentia. Something similar to the [audio slicer fun](https://mtg.github.io/essentia.js/examples/demos/onsets/public/) example from essentia.js.

Any help on this would be really appreciated. 

Thanks in advance

**Comment by Hikari-Tsai** [comment]
Onset detection and musical-structure segmentation are different tasks. The linked demo finds transient/event boundaries; it cannot label sections as intro, verse, chorus, or bridge.

Essentia can provide useful lower-level information:

- `SBic` can split a sequence of frame-level features into statistically homogeneous segments.
- novelty curves and peaks can provide candidate change points.
- tonal, timbral, rhythm and embedding features can be compared to find repeated sections.

However, those boundaries have no semantic labels. A practical structure-analysis pipeline is:

1. compute overlapping frame/patch embeddings or combined MFCC/chroma/rhythm features,
2. build a self-similarity matrix,
3. detect novelty peaks as boundaries,
4. cluster mutually similar segments to identify repeated sections, and
5. apply a separately trained model or heuristics to assign names such as “verse” and “chorus.”

`SBic` alone is useful for step 3 but will not know that two non-adjacent segments are both choruses. Therefore there is no single Essentia algorithm that returns the requested marker labels; it requires a music-structure model on top of Essentia features (or a dedicated pretrained structural-segmentation system).
